source/common/filesystem/win32/watcher_impl.cc

```diff
@@ -13,15 +13,15 @@ WatcherImpl::WatcherImpl(Event::Dispatcher& dispatcher, Api::Api& api)
   Api::SysCallIntResult result = os_sys_calls_.socketpair(AF_INET, SOCK_STREAM, IPPROTO_TCP, socks);
   ASSERT(result.rc_ == 0);
 
-  event_read_ = socks[0];
-  event_write_ = socks[1];
-  result = os_sys_calls_.setsocketblocking(event_read_, false);
+  read_handle_ = std::make_unique<Network::IoSocketHandleImpl>(socks[0], false, AF_INET);
```

Comment:
yeah this API is much easier. I tried to apply it to the rest of the watcher but it ended up being harder than expected because kqueue requires a raw fd